From fcf8db804e2b185fdbea868c3ed9d931c7a41d2f Mon Sep 17 00:00:00 2001
From: jinghua <jinghua@marvell.com>
Date: Tue, 26 Jan 2016 22:21:34 +0800
Subject: [PATCH 0742/1240] fix: dts: A3700: disable palladium dts compilation
 with Multi-fdt feature

	For multi-fdt feature, all the dtb files that are compiled will
	be linked into spl/u-boot, but A3700 spl is running in SRAM, so
	the size of spl could not be bigger than SRAM.

	right now the size of spl is very close to SRAM, and there is
	space for only two dtb files.
	So when Multi-FDT feature is included, these two dtb files that
	are compiled/linked into SPL and u-boot have to be the two for
	A3700 DB, and armada-lp-palladium.dtb would not be compiled.

	there is a JIRA bug about this issue: SYSTEMSW-2194, when it is
	fixed, this patch could be removed.

Change-Id: If0cc3e62c35351073b3cbfae130e74c82c043ce8
Signed-off-by: jinghua <jinghua@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27021
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 arch/arm/dts/Makefile | 17 +++++++++++++++--
 1 file changed, 15 insertions(+), 2 deletions(-)

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 00e8004..aece8d8 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -56,9 +56,22 @@ ifdef CONFIG_TARGET_ARMADA_LP
 ifndef CONFIG_DEVEL_BOARD
 dtb-y += armada-lp-customer0.dtb
 else
+# For multi-fdt feature, all the dtb files that are compiled will
+# be linked into spl/u-boot, but A3700 spl is running in SRAM, so
+# the size of spl could not be bigger than SRAM.
+# right now the size of spl is very close to SRAM, and there is
+# space for only two dtb files.
+# So when Multi-FDT feature is included, these two dtb files that
+# are compiled/linked into SPL and u-boot have to be the two for
+# A3700 DB, and armada-lp-palladium.dtb would not be compiled.
+# there is a JIRA bug about this issue: SYSTEMSW-2194, when it is
+# fixed, this patch could be removed.
 dtb-y += armada-lp-db.dtb \
-	 armada-lp-db-sgmii1.dtb \
-	 armada-lp-palladium.dtb
+	 armada-lp-db-sgmii1.dtb
+ifndef CONFIG_MULTI_DT_FILE
+dtb-y += armada-lp-palladium.dtb
+endif #CONFIG_MULTI_DT_FILE
+
 endif #CONFIG_DEVEL_BOARD
 endif #CONFIG_TARGET_ARMADA_LP
 
-- 
1.9.1

